runtime/runtime.h

```diff
@@ -5,13 +5,25 @@
 #include <stdint.h>
 #include <stdbool.h>
 #include <stdio.h>
-#include "ocl.h"
-#include "time_utils.h"
 
 typedef struct ContextImpl* Context;
 typedef struct KernelImpl* Kernel;
 typedef struct BufferImpl* Buffer;
 
+typedef struct {
+  size_t size;
+  void* value;
+} KernelArg;
```

Comment:
Why is this in `runtime.h`? It is useful for `launchKernel` which is defined by the backend. I would move this to the specific backends.

Reply:
Right now I reused ```struct KernelArg``` to pass arguments to a function which is 'kernel' in my case

Reply:
Moved ```struct KernelArg``` to Ocl specific include